marklogic-data-hub-spark-connector/src/main/java/com/marklogic/hub/spark/sql/sources/v2/DefaultSource.java

```diff
@@ -216,6 +214,16 @@ private DocumentMetadataHandle buildDocumentMetadata() {
     }
 
     private String initializeJob(StructType schema) {
+        String apiPath = "/marklogic-data-hub-spark-connector/initializeJob.api";
+        String scriptPath = "/marklogic-data-hub-spark-connector/initializeJob.sjs";
+        try {
+            if (endpointExists(apiPath)) {
+                loadEndpoint(scriptPath, apiPath);
+            }
+        } catch (IOException e) {
+            throw new IllegalArgumentException("Unable to load initializeJob Endpoint, cause: " + e.getMessage(), e);
```

Comment:
Just throw a RuntimeException here - IllegalArgumentException is for when the user provided some argument that's not valid, which isn't the case here.

Reply:
Done.